feeder/priceprovider/sources/bybit.go

```diff
@@ -0,0 +1,68 @@
+package sources
+
+import (
+	"encoding/json"
+	"github.com/NibiruChain/nibiru/x/common/set"
+	"github.com/NibiruChain/pricefeeder/types"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+const (
+	Bybit = "bybit"
+)
+
+var _ types.FetchPricesFunc = BybitPriceUpdate
+
+type BybitTicker struct {
+	Symbol string `json:"symbol"`
+	Price  string `json:"lastPrice"`
+}
+
+type Response struct {
+	Data struct {
+		List []BybitTicker `json:"list"`
+	} `json:"result"`
+}
+
+// BybitPriceUpdate returns the prices for given symbols or an error.
+// Uses BYBIT API at https://bybit-exchange.github.io/docs/v5/market/tickers.
+func BybitPriceUpdate(symbols set.Set[types.Symbol]) (rawPrices map[types.Symbol]float64, err error) {
+
+	url := "https://api.bybit.com/v5/market/tickers?category=spot"
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var response Response
+	err = json.Unmarshal(b, &response)
+	if err != nil {
+		return nil, err
+	}
+
+	rawPrices = make(map[types.Symbol]float64)
+
+	for _, ticker := range response.Data.List {
```

Comment:
Is there a way to grab only specific tickers instead of the entire array and filtering through? It would save a lot on network bandwidth and computation.

Reply:
According to their API, you need to include the 'symbol' parameter to specify the ticker. Here's the example of the request:

`GET /v5/market/tickers?symbol=BTCUSD`

Let me know please, if you want me to apply these changes. Thanks.

Reply:
Thanks. I rebased into https://github.com/NibiruChain/pricefeeder/pull/39 and gave you credit.